code/lib/cli-storybook/src/automigrate/fixes/consolidated-imports.ts

```diff
@@ -0,0 +1,215 @@
+
+  return hasChanges ? recast.print(ast).code : null;
+}
+
+export const transformPackageJsonFiles = async (files: string[], dryRun: boolean) => {
+  const errors: Array<{ file: string; error: Error }> = [];
+  const limit = pLimit(10);
+
+  await Promise.all(
+    files.map((file) =>
+      limit(async () => {
+        try {
+          const contents = await readFile(file, 'utf-8');
+          const transformed = transformPackageJson(contents);
+          if (!dryRun && transformed) {
+            await writeFile(file, transformed);
+          }
+        } catch (error) {
+          errors.push({ file, error: error as Error });
+        }
+      })
+    )
+  );
+
+  return errors;
+};
+
+export const transformImportFiles = async (files: string[], dryRun: boolean) => {
+  const errors: Array<{ file: string; error: Error }> = [];
+  const limit = pLimit(10);
+
+  await Promise.all(
+    files.map((file) =>
+      limit(async () => {
+        try {
+          const contents = await readFile(file, 'utf-8');
+          const transformed = transformImports(contents);
+          if (!dryRun && transformed) {
+            await writeFile(file, transformed);
+          }
+        } catch (error) {
+          errors.push({ file, error: error as Error });
+        }
+      })
+    )
+  );
+
+  return errors;
+};
+
+export const consolidatedImports: Fix<ConsolidatedOptions> = {
+  id: 'consolidated-imports',
+  versionRange: ['^8.0.0', '^9.0.0'],
+  check: async () => {
+    const projectRoot = getProjectRoot();
+    const packageJsonFiles = await globby(['**/package.json'], {
+      ...commonGlobOptions(''),
+      ignore: ['**/node_modules/**'],
+      cwd: projectRoot,
+    });
+
+    // check if any of the package.json files have consolidated packages
+    const hasConsolidatedDependencies = await Promise.all(
+      packageJsonFiles.map(async (file) => {
+        const contents = await readFile(file, 'utf-8');
+        const packageJson = JSON.parse(contents);
+        return (
+          Object.keys(packageJson.dependencies || {}).some((dep) => dep in consolidatedPackages) ||
+          Object.keys(packageJson.devDependencies || {}).some((dep) => dep in consolidatedPackages)
+        );
+      })
+    ).then((results) => results.some(Boolean));
+
+    if (!hasConsolidatedDependencies) {
+      return null;
+    }
+    return {
+      packageJsonFiles,
+    };
+  },
+  prompt: (result: ConsolidatedOptions) => {
+    return dedent`
+      Found package.json files that contain consolidated Storybook packages that need to be updated:
+      ${result.packageJsonFiles.map((file) => `- ${file}`).join('\n')}
+
+      These packages have been consolidated into the main storybook package and should be removed.
+      The main storybook package will be added to devDependencies if not already present.
+      
+      Would you like to:
+      1. Update these package.json files
+      2. Scan your codebase and update any imports from these consolidated packages
```

Comment:
Historically, these kind of file transformations were "migrations". Now we want to have them as "automigrations". When used as "migrations", users had the possibility to apply a different glob pattern for the file transformations. I followed this pattern in my PR: https://github.com/storybookjs/storybook/pull/30761 and I think it makes sense to also follow it here (prompting the user for a glob pattern before continuing)